In [16]:
# importing requried Libraries

import numpy as np
import pandas as pd


In [17]:
# Loading Dataset

purchase = pd.read_csv("purchase.csv")
transaction = pd.read_csv("transction.csv") 

In [18]:
purchase 

LYLTY_CARD_NBR               LIFESTAGE PREMIUM_CUSTOMER
0                1000   YOUNG SINGLES/COUPLES          Premium
1                1002   YOUNG SINGLES/COUPLES       Mainstream
2                1003          YOUNG FAMILIES           Budget
3                1004   OLDER SINGLES/COUPLES       Mainstream
4                1005  MIDAGE SINGLES/COUPLES       Mainstream
...               ...                     ...              ...
72632         2370651  MIDAGE SINGLES/COUPLES       Mainstream
72633         2370701          YOUNG FAMILIES       Mainstream
72634         2370751          YOUNG FAMILIES          Premium
72635         2370961          OLDER FAMILIES           Budget
72636         2373711   YOUNG SINGLES/COUPLES       Mainstream

[72637 rows x 3 columns]

In [19]:
transaction

DATE  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR   
0       43390          1            1000       1         5  \
1       43599          1            1307     348        66   
2       43605          1            1343     383        61   
3       43329          2            2373     974        69   
4       43330          2            2426    1038       108   
...       ...        ...             ...     ...       ...   
264831  43533        272          272319  270088        89   
264832  43325        272          272358  270154        74   
264833  43410        272          272379  270187        51   
264834  43461        272          272379  270188        42   
264835  43365        272          272380  270189        74   

                                       PROD_NAME  PROD_QTY  TOT_SALES  
0         Natural Chip        Compny SeaSalt175g         2        6.0  
1                       CCs Nacho Cheese    175g         3        6.3  
2         Smiths Crinkle Cut  Chips Chicken 170g         2        2.9  
3         Smiths Chip Thinly  S/Cream&Onion 175g         5       15.0  
4       Kettle Tortilla ChpsHny&Jlpno Chili 150g         3       13.8  
...                                          ...       ...        ...  
264831   Kettle Sweet Chilli And Sour Cream 175g         2       10.8  
264832             Tostitos Splash Of  Lime 175g         1        4.4  
264833                  Doritos Mexicana    170g         2        8.8  
264834   Doritos Corn Chip Mexican Jalapeno 150g         2        7.8  
264835             Tostitos Splash Of  Lime 175g         2        8.8  

[264836 rows x 8 columns]

In [20]:
# Merge transaction data with customer purchase behavior to make single data Frame

merged = transaction.merge(purchase, on="LYLTY_CARD_NBR")



In [21]:
merged

DATE  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR   
0       43390          1            1000       1         5  \
1       43599          1            1307     348        66   
2       43414          1            1307     346        96   
3       43533          1            1307     347        54   
4       43605          1            1343     383        61   
...       ...        ...             ...     ...       ...   
264831  43533        272          272319  270088        89   
264832  43325        272          272358  270154        74   
264833  43410        272          272379  270187        51   
264834  43461        272          272379  270188        42   
264835  43365        272          272380  270189        74   

                                      PROD_NAME  PROD_QTY  TOT_SALES   
0        Natural Chip        Compny SeaSalt175g         2        6.0  \
1                      CCs Nacho Cheese    175g         3        6.3   
2                WW Original Stacked Chips 160g         2        3.8   
3                             CCs Original 175g         1        2.1   
4        Smiths Crinkle Cut  Chips Chicken 170g         2        2.9   
...                                         ...       ...        ...   
264831  Kettle Sweet Chilli And Sour Cream 175g         2       10.8   
264832            Tostitos Splash Of  Lime 175g         1        4.4   
264833                 Doritos Mexicana    170g         2        8.8   
264834  Doritos Corn Chip Mexican Jalapeno 150g         2        7.8   
264835            Tostitos Splash Of  Lime 175g         2        8.8   

                     LIFESTAGE PREMIUM_CUSTOMER  
0        YOUNG SINGLES/COUPLES          Premium  
1       MIDAGE SINGLES/COUPLES           Budget  
2       MIDAGE SINGLES/COUPLES           Budget  
3       MIDAGE SINGLES/COUPLES           Budget  
4       MIDAGE SINGLES/COUPLES           Budget  
...                        ...              ...  
264831   YOUNG SINGLES/COUPLES          Premium  
264832   YOUNG SINGLES/COUPLES          Premium  
264833   YOUNG SINGLES/COUPLES          Premium  
264834   YOUNG SINGLES/COUPLES          Premium  
264835   YOUNG SINGLES/COUPLES          Premium  

[264836 rows x 10 columns]

In [22]:
# Identifying top 3 most profitable products

top_products = (transaction.groupby(["PROD_NBR", "PROD_NAME"])["TOT_SALES"].sum().sort_values(ascending=False))

In [23]:
top_products.head(3)

PROD_NBR  PROD_NAME                               
4         Dorito Corn Chp     Supreme 380g            40352.0
14        Smiths Crnkle Chip  Orgnl Big Bag 380g      36367.6
16        Smiths Crinkle Chips Salt & Vinegar 330g    34804.2
Name: TOT_SALES, dtype: float64

In [24]:
# Identifying the most loyal customer segment

loyal_customers = (merged.groupby(["LIFESTAGE", "PREMIUM_CUSTOMER"])["TOT_SALES"] .sum().sort_values(ascending=False))


In [25]:
loyal_customers

LIFESTAGE               PREMIUM_CUSTOMER
OLDER FAMILIES          Budget              168363.25
YOUNG SINGLES/COUPLES   Mainstream          157621.60
RETIREES                Mainstream          155677.05
YOUNG FAMILIES          Budget              139345.85
OLDER SINGLES/COUPLES   Budget              136769.80
                        Mainstream          133393.80
                        Premium             132263.15
RETIREES                Budget              113147.80
OLDER FAMILIES          Mainstream          103445.55
RETIREES                Premium              97646.05
YOUNG FAMILIES          Mainstream           92788.75
MIDAGE SINGLES/COUPLES  Mainstream           90803.85
YOUNG FAMILIES          Premium              84025.50
OLDER FAMILIES          Premium              81958.40
YOUNG SINGLES/COUPLES   Budget               61141.60
MIDAGE SINGLES/COUPLES  Premium              58432.65
YOUNG SINGLES/COUPLES   Premium              41642.10
MIDAGE SINGLES/COUPLES  Budget           

In [30]:
# Displaying results

print("Top 3 Most Profitable Products:\n", top_products)
print("\n Most Loyal Customers:\n", loyal_customers)


Top 3 Most Profitable Products:
 PROD_NBR  PROD_NAME                               
4         Dorito Corn Chp     Supreme 380g            40352.0
14        Smiths Crnkle Chip  Orgnl Big Bag 380g      36367.6
16        Smiths Crinkle Chips Salt & Vinegar 330g    34804.2
102       Kettle Mozzarella   Basil & Pesto 175g      34457.4
7         Smiths Crinkle      Original 330g           34302.6
                                                       ...   
92        WW Crinkle Cut      Chicken 175g             4702.2
95        Sunbites Whlegrn    Crisps Frch/Onin 90g     4600.2
72        WW Crinkle Cut      Original 175g            4532.2
35        Woolworths Mild     Salsa 300g               4234.5
76        Woolworths Medium   Salsa 300g               4050.0
Name: TOT_SALES, Length: 114, dtype: float64

 Most Loyal Customers:
 LIFESTAGE               PREMIUM_CUSTOMER
OLDER FAMILIES          Budget              168363.25
YOUNG SINGLES/COUPLES   Mainstream          157621.60
RETIREES       

# Summary of findings

Top 3 Most Profitable Products:


Dorito Corn Chp Supreme 380g                40352.0

Smiths Crnkle Chip  Orgnl Big Bag 380g      36367.6

Smiths Crinkle Chips Salt & Vinegar 330g    34804.2

Most Loyal Customers :

OLDER FAMILIES       

YOUNG SINGLES/COUPLES 

RETIREES             

YOUNG FAMILIES      

OLDER SINGLES/COUPLES  
    
    
